storage/remote/client.go

```diff
@@ -198,11 +204,33 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 		err = errors.Errorf("server returned HTTP status %s: %s", httpResp.Status, line)
 	}
 	if httpResp.StatusCode/100 == 5 {
-		return RecoverableError{err}
+		return RecoverableError{err, defaultBackoff}
+	}
+	if httpResp.StatusCode == 429 {
+		return RecoverableError{err, retryAfterDuration(httpResp.Header.Get("Retry-After"))}
 	}
 	return err
 }
 
+// retryAfterDuration returns the duration for the Retry-After header. In case of any errors, it
+// returns the defaultBackoff as if the header was never supplied.
+func retryAfterDuration(t string) model.Duration {
+	parsedDuration, err := time.Parse(time.RFC1123, t)
+	if err == nil {
+		if parsedDuration.Before(time.Now()) {
+			return invalidRetryAfter
```

Comment:
If we check for < 0 later on we do not need this.

Reply:
We can do this way as well or have a unit test for this function (which seems more reliable). For now, I am going with the later.

Reply:
Yeah, the more I think about it the more I would prefer the check for < 0 rather than depending on -1. A whole lot more invalid cases are covered with that check in the case of future changes.